aqt/metadata.py

```diff
@@ -270,24 +260,45 @@ def __str__(self) -> str:
         )
 
 
-class Table:
-    def __init__(self, head: List[str], rows: List[List[str]], max_width: int = 0):
-        # max_width is set to 0 by default: this disables wrapping of text table cells
-        self.head = head
-        self.rows = rows
-        self.max_width = max_width
+class ToolData:
+    """A data class hold tool details."""
+
+    head = [
+        "Tool Variant Name",
+        "Version",
+        "Release Date",
+        "Display Name",
+        "Description",
+    ]
+
+    def __init__(self, tool_data):
+        self.tool_data = tool_data
 
     def __format__(self, format_spec) -> str:
-        if format_spec == "":
-            table = Texttable(max_width=self.max_width)
-            table.set_deco(Texttable.HEADER)
-            table.header(self.head)
-            table.add_rows(self.rows, header=False)
-            return table.draw()
-        elif format_spec == "s":
+        if format_spec == "{s}":
             return str(self)
+        if format_spec == "":
+            max_width: int = 0
         else:
-            raise ValueError()
+            match = re.match(r"\{(.*):(\d*)t\}", format_spec)
+            if match:
+                g = match.groups()
+                max_width = 0 if g[1] == "" else int(g[1])
+            else:
+                raise ValueError("Wrong format")
+        table = Texttable(max_width=max_width)
+        table.set_deco(Texttable.HEADER)
+        table.header(self.head)
+        table.add_rows(self.rows, header=False)
+        return table.draw()
+
+    @property
+    def rows(self):
+        keys = ("Version", "ReleaseDate", "DisplayName", "Description")
+        return [
+            [name, *[content[key] for key in keys]]
+            for name, content in self.tool_data.items()
+        ]
 
 
 class ListCommand:
```

Comment:
It may be better to change the name to express what it is doing.

Now `ListCommand` class has only a factory functions to retrieve Update.xml and produce metadata(`List[str]`, `Versions` or `ToolData`).
Here is `ListCommand` as a factory class, `ToolData` is a data class, and `show_list` function is a consumer.

There is `QtArchives` as a factory class, `List[QtPackages]` is a data, and `call_installer` function is a consumer.

There are more spaces to improve a structure of classes.

Reply:
I agree. Maybe we could rename `ListCommand` to something like `MetadataFactory`? Maybe `ListCommand.action()` could be renamed `MetadataFactory.getMetadata()`?

Reply:
done